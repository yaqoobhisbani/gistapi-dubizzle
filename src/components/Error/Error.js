import React from "react";
import styled from "styled-components";
import Octicon from "react-octicon";

const Error = ({ message }) => {
  return (
    <Wrapper>
      <Octicon name="stop" style={icon} data-testid="stop-icon" />
      <Message>
        {message ? message : "Something went wrong! Please try again later."}
      </Message>
    </Wrapper>
  );
};

const Wrapper = styled.div`
  display: flex;
  align-items: center;
  flex-direction: column;
  margin: 16px;
`;

const Message = styled.p`
  font-size: 16px;
  margin-top: 8px;
`;

const icon = { fontSize: 48, color: "red" };

export default Error;
