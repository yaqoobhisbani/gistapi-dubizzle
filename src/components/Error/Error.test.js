import React from "react";
import { render } from "@testing-library/react";
import Error from "./Error";

describe("Error", () => {
  test("renders default message when no message prop is passed", () => {
    const { getByText } = render(<Error />);
    const messageElement = getByText(
      "Something went wrong! Please try again later."
    );
    expect(messageElement).toBeInTheDocument();
  });

  test("renders custom message when message prop is passed", () => {
    const customMessage = "Oops, something unexpected happened!";
    const { getByText } = render(<Error message={customMessage} />);
    const messageElement = getByText(customMessage);
    expect(messageElement).toBeInTheDocument();
  });

  test("renders stop icon", () => {
    const { getByTestId } = render(<Error />);
    const iconElement = getByTestId("stop-icon");
    expect(iconElement).toBeInTheDocument();
  });

  test("matches snapshot", () => {
    const { container } = render(<Error />);
    expect(container).toMatchSnapshot();
  });
});
