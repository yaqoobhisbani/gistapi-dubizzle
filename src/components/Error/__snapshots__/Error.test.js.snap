// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error matches snapshot 1`] = `
<div>
  <div
    class="sc-beqWaB eoxPnZ"
  >
    <span
      class="octicon octicon-stop"
      data-testid="stop-icon"
      style="font-size: 48px; color: red;"
    />
    <p
      class="sc-gueYoa dDXANv"
    >
      Something went wrong! Please try again later.
    </p>
  </div>
</div>
`;
